### PR TITLE
DBZ-1185 Support for SERIAL type alias

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.jboss.org/browse/DBZ-1185

### CODE REVIEW

**Comment by jpechane** [comment]
Upstream issue - https://github.com/antlr/grammars-v4/pull/1377

**Comment by jpechane** [comment]
The test is failing because legacy parser does not support this functionality. Is it time drop it from CI?

**Comment by gunnarmorling** [comment]
>  Is it time drop it from CI?

No, I don't think we should do this yet. After all, it's still available functionality (only deprecated). We should wait with that until it's removed in 0.10. Can we disable the test for the time being?

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/listener/ColumnDefinitionParserListener.java [hunk 1]
```diff
@@ -37,6 +38,8 @@
     private final DataTypeResolver dataTypeResolver;
     private final TableEditor tableEditor;
     private ColumnEditor columnEditor;
+    private boolean uniqueColumn;
+    private Optional<Boolean> optionalColumn;
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/listener/ColumnDefinitionParserListener.java]: There's not much point in using `Optional` as a field. Simply use `Boolean`.

Comment (jpechane) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/listener/ColumnDefinitionParserListener.java]: @gunnarmorling I've initially used the `Boolean` but I belive that `Optional` expess three state semantics not-set/false/true better than `null`/false/true.
`Optional` explicitly forces you to check, `null` checking is only optional

Comment (gunnarmorling) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/listener/ColumnDefinitionParserListener.java]: I dare to disagree :) Keep in mind `Optional` was only foreseen as a method return type, e.g. see what Stuart Marks has to say on the issue: https://stackoverflow.com/questions/23454952/uses-for-optional

**Comment by jpechane** [comment]
@gunnarmorling I've added a new test rule and annotation to skip the failing tests when legacy parser is in use

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/listener/ColumnDefinitionParserListener.java (hunk 1 above)
Comment (jpechane) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/listener/ColumnDefinitionParserListener.java]: @gunnarmorling ok, you are the boss ;-) - changed to `Boolean`.

BTW let me try (https://www.logicallyfallacious.com/tools/lp/Bo/LogicalFallacies/21/Appeal-to-Authority) as counterargument and point you to https://twitter.com/mariofusco/status/518729164384645120 :-)

And a chapter in https://livebook.manning.com/#!/book/modern-java-in-action/chapter-11/1

Comment (gunnarmorling) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/listener/ColumnDefinitionParserListener.java]: LOL, who's that Fusco guy ;)

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/listener/ColumnDefinitionParserListener.java [hunk 2]
```diff
@@ -62,33 +64,44 @@ public Column getColumn() {
...
+        uniqueColumn = false;
+        optionalColumn = null;
         resolveColumnDataType(ctx.dataType());
         super.enterColumnDefinition(ctx);
     }
 
     @Override
-    public void enterUniqueKeyColumnConstraint(MySqlParser.UniqueKeyColumnConstraintContext ctx) {
-        if (!tableEditor.hasPrimaryKey()) {
+    public void exitColumnDefinition(MySqlParser.ColumnDefinitionContext ctx) {
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/listener/ColumnDefinitionParserListener.java]: Don't we have to reset the `uniqueColumn` and `optionalColumn` when exiting the column definition?

Comment (jpechane) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/listener/ColumnDefinitionParserListener.java]: It is reset in `enterColumnDefinition`

Comment (gunnarmorling) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/listener/ColumnDefinitionParserListener.java]: Ah yes, thanks!

**Comment by gunnarmorling** [comment]
Rebased and applied. Thanks, @jpechane!

**Pull request merged**